infra/src/main/java/com/redhat/service/bridge/infra/models/processors/ProcessorDefinition.java

```diff
@@ -0,0 +1,71 @@
+package com.redhat.service.bridge.infra.models.processors;
+
+import java.util.Objects;
+import java.util.Set;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.redhat.service.bridge.infra.models.actions.BaseAction;
+import com.redhat.service.bridge.infra.models.filters.BaseFilter;
+
+public class ProcessorDefinition {
```

Comment:
The idea of "schemaless" for processor definition is intended to be for the whole chain of processing: no Entity but also not "DTO" otherwise we are just moving the problem from DB to Manager code.
I think the conversion to an actual bean should happen (if necessary) only at Shard/Executor level while in the manager you can use JsonNode or similar

Reply:
my2c: the manager already serializes the ProcessorDefinition when the request comes (also for validation), so it already has a dependency to such structure. 
I'd keep using `ProcessorDefinition` instead of JsonNode in the DTOs, it's much simpler imo.

Reply:
Let me spend a couple of words on this topic: I think making use of the JsonNode in the manager is not enough. Let's assume we have a breaking change in the `ProcessorDefinition` from
```json
{
"filters": [ {"type": "StringEquals", "value": "jacopo", "key": "data.name"} ],
"transformationTemplate": "",
"actions": []
}
```
to
```json
"filters": [ {"type": "StringEquals", "filterValue": "jacopo", "key": "data.name"} ],
"transformationTemplate":"",
"actions": []
```
i.e. `value` turned into `filterValue` (we can think about more realistic examples, but it's just to give the idea). The manager can simply store the (unstructured) data into the `processor_definition` column, and return it to every shard as it is. 

The problem is that now the shards must know how they have to deserialize the object. In theory with the endpoint `/api/v1/shard/processors` it might even get processors with different schemas!

If we have a breaking change we have different options
1) we have to migrate the existing data
2) we have to store the new data in a different column (or add a version property in the data) so that the manager can expose the processors depending on their schema

Reply:
As per some of the issues raised by @r00ta , I would expect any update to our API structures to be rolled out gradually and by definition we'd have to support N-1 compatibility in the Manager for a period of time. So for example changing the data structure for the ProcessorDefinition, I'd expect something like this as a flow of deployment (this would happen over multiple deployments into prod)

1. Create the new schema and add a new column to the DB. Write a migration to copy all existing data to the new version and update the Manager to dual write into both columns
2. Update the  the API on the Manager that allows retrieval of either old or new version of the data structure
3. Update the shard to be able to work with new version of the data structure and deploy it to a subset of our shards
4. Monitor things for a bit and assuming all good, rollout to all shards
5. Once we're happy, remove the dual write in the Manager and drop the legacy column

Reply:
Following our sync call on this topic, I implemented the `definition` field as `JsonNode` only in the model and kept every other DTO / Request / Response as it is. We'll discuss the plan for versioning and data migration separately.